opcua/server/address_space.py

```diff
@@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
     def _add_node(self, item, user):
         result = ua.AddNodesResult()
 
-        if item.RequestedNewNodeId in self._aspace:
+        # Check if the null NodeId is given
+        if item.RequestedNewNodeId.is_null():
+            self.logger.debug("RequestedNewNodeId is null")
+            if item.BrowseName.NamespaceIndex:
```

Comment:
Now I am back,....what about extending the specification a bit and
creating  new nodeid every time the
identifier is 0 or ""? We should not anyway create such a a NodeID and it
allows us to specify a namespace while letting the server create the nodeid.
We could add a method has_null_identifier() to NodeId. Does it sound
reasonable? We also need to documente this behaviour in add_node method of
InternalServer


On Wed, 23 Mar 2016 at 11:22 Denis Štogl <notifications@github.com> wrote:

> In opcua/server/address_space.py
> <https://github.com/FreeOpcUa/python-opcua/pull/149#discussion_r57136311>:
>
> > @@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
> >      def _add_node(self, item, user):
> >          result = ua.AddNodesResult()
> >
> > -        if item.RequestedNewNodeId in self._aspace:
> > +        # Check if the null NodeId is given
> > +        if item.RequestedNewNodeId.is_null():
> > +            self.logger.debug("RequestedNewNodeId is null")
> > +            if item.BrowseName.NamespaceIndex:
>
> Have a fun :)
>
> After the analysis of AddNode Service I found BrowseName the most
> propriate, why that would be wrong? On the end of the day you are mostly
> searching for nodes through BrowseName which is contained of NamespaceIndex
> and Name...
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeOpcUa/python-opcua/pull/149/files/a6be8b95b418d4abec719247991a80265bbcec86#r57136311>
>